core/server/data/sitemap/base-generator.js

```diff
@@ -24,10 +25,16 @@ function BaseSiteMapGenerator() {
 }
 
 _.extend(BaseSiteMapGenerator.prototype, {
+    dataEvents: events,
     init: function () {
-        return this.refreshAll();
+        var self = this;
+        return this.refreshAll().then(function () {
+            return self.bindEvents();
+        });
     },
 
+    bindEvents: function () {},
```

Comment:
Could use `_.noop` here for that super-fast-micro-optimization-bleeding-metal-iwatch-3.0 speed.